### PR TITLE
samples: openthread: support nrf5340dk_nrf5340_cpuapp_ns target

### PR DESCRIPTION
**Pull request by Damian-Nordic**
Allow to build openthread samples: cli, coap_client and coap_server for the nrf5340dk_nrf5340_cpuapp_ns build target:
* Add boards/nrf5340dk_nrf5340_cpuapp_ns.conf which enables BUILD_WITH_TFM option. Note that PSA Crypto API is still not used as it depends on the Zephyr upmerge.
* Update documentation.
* Update sample.yaml files.

### CODE REVIEW

File: samples/openthread/coap_client/sample.yaml [hunk 1]
```diff
@@ -4,23 +4,25 @@ sample:
 tests:
   samples.openthread.coap_client:
     build_only: true
-    platform_allow: nrf5340dk_nrf5340_cpuapp nrf52840dk_nrf52840 nrf52833dk_nrf52833 nrf21540dk_nrf52840
+    platform_allow: nrf52840dk_nrf52840 nrf52833dk_nrf52833 nrf5340dk_nrf5340_cpuapp nrf5340dk_nrf5340_cpuapp_ns
```
Comment (MarekPorwisz) [inline samples/openthread/coap_client/sample.yaml:7]: why did you remove the nrf21540dk_nrf52840 (also in other yaml files)

Comment (Damian-Nordic) [reply samples/openthread/coap_client/sample.yaml:7]: Right, the mergetool got it wrong after a rebase.

File: samples/openthread/cli/boards/nrf5340dk_nrf5340_cpuapp_ns.conf [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Enable Trusted Firmware M
+CONFIG_BUILD_WITH_TFM=y
+
+# Disable FPU since FP_HARD_ABI cannot be used with TF-M, and softfp-float
+# cryptographic libraries are unavailable.
+CONFIG_FPU=n
```
Comment (MarekPorwisz) [inline samples/openthread/cli/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:12]: Maybe we should discuss some possible issues. This requires another set of pre-built libraries. We would like to switch to sources certification in 1.8.0 but the final decision is not on Nordic. I'm afraid this may lead to some late issues discovery.
Also will this be covered by testing?

Comment (Damian-Nordic) [reply samples/openthread/cli/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:12]: I just talked about this with @SebastianBoe. Disabling FPU is a temporary workaround and it won't be needed eventually.

Comment (SebastianBoe) [reply samples/openthread/cli/boards/nrf5340dk_nrf5340_cpuapp_ns.conf:12]: See https://projecttools.nordicsemi.no/jira/browse/NCSDK-11600

**Review by SebastianBoe — APPROVED** [review]

**Comment by emob-nordic** [comment]
You should add entry in README files for each sample you add support for ns variant.
Currently there is no `thingy53_nrf5340_cpuapp_ns` defined in  `/includes/sample_board_rows.txt`. Im about to add this here: https://github.com/nrfconnect/sdk-nrf/pull/5727.

If the sample supports thingy53 secure and non secure you should add `thingy53_nrf5340_cpuapp_and_cpuapp_ns` in the table build from `/includes/sample_board_rows.txt`

**Comment by emob-nordic** [comment]
> You should add entry in README files for each sample you add support for ns variant. Currently there is no `thingy53_nrf5340_cpuapp_ns` defined in `/includes/sample_board_rows.txt`. Im about to add this here: #5727.
> 
> If the sample supports thingy53 secure and non secure you should add `thingy53_nrf5340_cpuapp_and_cpuapp_ns` in the table build from `/includes/sample_board_rows.txt`

Forget about it, sorry. I just realized you are adding support for the DK, not thingy. You should add the entry anyway but for the DK. You don't need to wait for the #5727.

**Review by LuDuda — APPROVED** [review]
LGTM :+1: Just remember to add the "_ns" variant to the samples doc.

File: samples/openthread/cli/sample.yaml [hunk 3]
```diff
@@ -22,7 +23,7 @@ tests:
       - nrf21540dk_nrf52840
   samples.openthread.cli.usb:
     build_only: true
-    platform_allow: nrf5340dk_nrf5340_cpuapp nrf52840dk_nrf52840 nrf52840dongle_nrf52840 nrf52833dk_nrf52833 nrf21540dk_nrf52840
+    platform_allow: nrf5340dk_nrf5340_cpuapp nrf5340dk_nrf5340_cpuapp_ns nrf52840dk_nrf52840 nrf52840dongle_nrf52840 nrf52833dk_nrf52833 nrf21540dk_nrf52840
     tags: ci_build
     extra_args: OVERLAY_CONFIG=overlay-usb.conf
     integration_platforms:
```
Comment (MarekPorwisz) [inline samples/openthread/cli/sample.yaml:29]: Is it a mistake or this section does not need to be updated (I do not remember what this section does :( )

**Review by MarekPorwisz — APPROVED** [review]
Looks good, but I would prefer this to be merged after CI is able to test _ns configuration

**Review by wiba-nordic — APPROVED** [review]
Giving this the tech writer approval since there are small docs changes.

File: samples/openthread/cli/sample.yaml (hunk 3 above)
Comment (Damian-Nordic) [reply samples/openthread/cli/sample.yaml:29]: I enabled the build target for all configurations.

File: samples/openthread/cli/boards/nrf5340dk_nrf5340_cpuapp_ns.conf (hunk 2 above)
Comment (Damian-Nordic) [reply samples/openthread/cli/boards/nrf5340dk_nrf5340_cpuapp_ns.conf]: We don't need to disable FPU anymore.

**Review by LuDuda — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Comment by Damian-Nordic** [comment]
I manually verified the change:
* tested that the CLI sample can successfully connect and exchanges message with OTBR
* tested that coap_client on nrf5340 (ns) can talk to coap_server on nrf52840 and the other way around.

We agreed that automated tests will be enabled later.
